pdns/iputils.cc

```diff
@@ -136,6 +136,25 @@ int SSetsockopt(int sockfd, int level, int opname, int value)
   return ret;
 }
 
+void setSocketIgnorePMTU(int sockfd)
+{
+#ifdef IP_PMTUDISC_OMIT
+  /* Linux 3.15+ has IP_PMTUDISC_OMIT, which discards PMTU information to prevent
+     poisoning, but still allows fragmentation if the packet size exceeds the
+     outgoing interface MTU, which is good.
+  */
+  try {
+    SSetsockopt(sockfd, IPPROTO_IP, IP_MTU_DISCOVER, IP_PMTUDISC_OMIT);
+    return;
+  }
+  catch(const std::exception& e) {
+    /* failed, let's try IP_PMTUDISC_DONT instead */
+  }
+#endif /* IP_PMTUDISC_OMIT */
+
+  /* IP_PMTUDISC_DONT disables Path MTU discovery */
+  SSetsockopt(sockfd, IPPROTO_IP, IP_MTU_DISCOVER, IP_PMTUDISC_DONT);
```

Comment:
We should check that `IP_MTUDISC_DONT` is defined. Note that unbound set the don't fragment flag on systems that don't have `IP_MTUDISC_DONT`: https://github.com/NLnetLabs/unbound/blob/master/services/listen_dnsport.c#L571

Reply:
Bind is doing the same: https://gitlab.isc.org/isc-projects/bind9/blob/master/lib/isc/unix/socket.c#L2396

Reply:
Added the missing check for `IP_MTUDISC_DONT`. I'm still undecided about setting DF=1.